Reflect.ts

```diff
@@ -1344,10 +1344,7 @@ module Reflect {
     // naive WeakMap shim
     function CreateWeakMapPolyfill() {
         const UUID_SIZE = 16;
-        const isNode = typeof global !== "undefined" &&
-            typeof module === "object" &&
-            typeof module.exports === "object" &&
-            typeof require === "function";
+        const isNode = Object.prototype.toString.call(global.process) === '[object process]';

```

Comment:
This would throw in the browser when `global` is undefined. It probably should read:

```ts
const isNode = typeof global !== "undefined" && Object.prototype.toString.call(global.process) === '[object process]';
```